src/rgw/rgw_user.cc

```diff
@@ -2462,8 +2465,11 @@ int RGWUserAdminOp_Subuser::remove(RGWRados *store, RGWUserAdminOpState& op_stat
 
 
   ret = user.subusers.remove(op_state, NULL);
-  if (ret < 0)
+  if (ret < 0) {
+    if (ret == -ENOENT)
+      ret = -ERR_NO_SUCH_SUBUSER;
```

Comment:
@joscollin 
Yes, I missed your point in the first comment, sorry. I was hoping to follow the code path just like the ones of `RGWUser`, narrow the returning of these specific error codes only to the frontend part. After rechecking `rgw_user.cc` and `rgw_admin.cc`, I think you are right, it won't conflict and is simpler to returning `ERR_NO_SUCH_SUBUSR`